camelot/view/controls/delegates/customdelegate.py

```diff
@@ -134,28 +136,34 @@ def __init__(self, parent=None, editable=True, **kwargs):
         self._width = self._font_metrics.averageCharWidth() * 20
 
     @classmethod
-    def get_standard_item(cls, locale, value, field_attributes_values):
+    def get_standard_item(cls, locale, model_context):
         """
         This method is used by the proxy to convert the value of a field
         to the data for the standard item model.  The result of this call can be
         used by the methods of the delegate.
 
         :param locale: the `QLocale` to be used to display locale dependent values
-        :param value: the value of the field on the object
-        :param field_attributes_values: the values of the field attributes on the
-           object
+        :param model_context: a FieldActionModelContext object
         
         :return: a `QStandardItem` object
         """
+        routes = model_context.field_attributes.get('action_routes')
+        states = []
+        for action in model_context.field_attributes.get('actions', []):
+            state = action.get_state(model_context)
+            states.append(dataclasses.asdict(state))
+        action_states = dict(zip(routes, states))
```

Comment:
De vfinance.test.test_connector.test_nbb:CentralCreditRegisterCase.test_create_request bleef nog hangen omdat er soms meer states dan action_routes zijn. Er stond hier een assertion die dit checkte. Is het een probleem als er niet evenveel action_states en action_routes zijn?

Reply:
dat is toch wel verdacht, kan je die assert een message laten meegeven zodat we weten waarover dit gaat.
zaken waar ik zo op het 1e zicht aan denk : 
- dat input scherm is mogelijk een beetje speciaal omwille van de types requests
- mogelijk w er bij het maken van screenshots een totaal anders scherm getriggerd
- mogelijk w er ergens per ongeluk een 'invariabele' lijst toch aangepast

@jeffh92 : als jij direct denkt aan iets speciaals bij die requests ...

Reply:
Mja, het enige wat ik zo direct kan bedenken dat er speciaal is aan die create action is dat in tegenstelling tot de algemene One2ManyAddNewObject actions die met facade klasses omgaat bij het openen van de form, er hier de request facades met een blocking change object worden geopend.

Reply:
De action_routes en states zijn:

```
[
    ('admin', '9', 'LoanAccount', 'fields', 'repayments', 'actions', 'add_object'),
    ('admin', '9', 'LoanAccount', 'fields', 'repayments', 'actions', 'remove_selection')
]
[
    {'verbose_name': _('Add'), 'icon': {'name': 'plus', 'pixmap_size': 32}, 'tooltip': 'Voeg toe', 'enabled': False, 'visible': True, 'notification': False, 'modes': []}, 
    {'verbose_name': _('Remove'), 'icon': {'name': 'minus', 'pixmap_size': 32}, 'tooltip': 'Verwijder', 'enabled': False, 'visible': True, 'notification': False, 'modes': []}, 
    {'verbose_name': _('Maak rappel'), 'icon': None, 'tooltip': None, 'enabled': True, 'visible': True, 'notification': False, 'modes': []}
]
```

Die laatste actie w toegevoegd in vfinance/model/hypo/rappel_brief.py:228:

```
        def get_related_toolbar_actions(self, toolbar_area, direction):
            from .admin import CreateReminder
            actions = InvoiceItem.Admin.get_related_toolbar_actions(self, toolbar_area, direction)
            actions.append( CreateReminder() )
            return actions
```